cmake/Platform/Common/runtime_dependencies_common.cmake.in

```diff
@@ -9,11 +9,18 @@
 cmake_policy(SET CMP0012 NEW) # new policy for the if that evaluates a boolean out of "if(NOT ${same_location})"
 
 function(ly_copy source_file target_directory)
-    get_filename_component(target_filename "${source_file}" NAME)
-    cmake_path(COMPARE "${source_file}" EQUAL "${target_directory}/${target_filename}" same_location)
+    cmake_path(GET source_file FILENAME target_filename)
+    cmake_path(APPEND target_file "${target_directory}" "${target_filename}")
+    cmake_path(COMPARE "${source_file}" EQUAL "${target_file}" same_location)
     if(NOT ${same_location})
-        file(LOCK ${target_directory}/${target_filename}.lock GUARD FUNCTION TIMEOUT 300)
-        if("${source_file}" IS_NEWER_THAN "${target_directory}/${target_filename}")
+        file(LOCK ${target_file}.lock GUARD FUNCTION TIMEOUT 300)
+        file(SIZE "${source_file}" source_file_size)
+        if(EXISTS "${target_file}")
+            file(SIZE "${target_file}" target_file_size)
+        else()
+            set(target_file_size 0)
+        endif()
+        if((NOT source_file_size EQUAL target_file_size) OR "${source_file}" IS_NEWER_THAN "${target_directory}/${target_filename}")
```

Comment:
`"${target_directory}/${target_filename}"` in the `if` seems it can be reduced to`"${target_file}"`

Same with the `file(TOUCH_NOCREATE ${target_directory}/${target_filename})` below.

Reply:
will simplify all these thanks